drivers/include/bme680.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2019 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_bme680 BME680 Temperature/Humidity/Pressure/Gas sensor
+ * @ingroup     drivers_sensors
+ * @brief       Driver for the Bosch BME680  sensor
+ *
+ * @{
+ * @file
+ * @brief       Interface definition for the Bosch BME680 sensor
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ */
+
+#ifndef BME680_H
+#define BME680_H
+
+#include "periph/i2c.h"
+#include "periph/spi.h"
+
+#include "bme680_internal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+#ifdef MODULE_BME680_FP
+#define BME680_FLOAT_POINT_COMPENSATION
+#endif
+
+/**
+ * @brief BME680 I2C parameters
+ */
+typedef struct {
+  i2c_t dev;                    /**< I2C device which is used */
+  uint8_t addr;                 /**< I2C address */
+} bme680_intf_i2c_t;
+
+/**
+ * @brief BME680 SPI parameters
+ */
+typedef struct {
+    spi_t dev;                  /**< SPI device which is used */
+    gpio_t nss_pin;             /**< Chip Select pin */
+} bme680_intf_spi_t;
+
+/**
+ * @brief BME680 Hardware interface parameters union
+ */
+typedef union {
+  bme680_intf_i2c_t i2c;        /**< I2C specific interface parameters */
+  bme680_intf_spi_t spi;        /**< SPI specific interface parameters */
+} bme680_intf_t;
+
+/**
+ * @brief   BME680 device initialization parameters
+ */
+typedef struct {
+    uint8_t ifsel;              /**< Interface selection */
+    uint8_t temp_os;            /**< Temperature oversampling */
+    uint8_t hum_os;             /**< Humidity oversampling */
+    uint8_t pres_os;            /**< Pressure oversampling */
+    uint8_t filter;             /**< IIR filter coefficient */
+    uint8_t gas_measure;        /**< Enable gas measurement */
+    uint16_t heater_dur;        /**< Heater duration in ms */
+    uint16_t heater_temp;       /**< Heater temperature in °C */
+    uint8_t power_mode;         /**< Power mode (sleep or forced) */
```

Comment:
I'm not sure, whether the `power_mode` makes sense as part of parameters. The only setting that is realized by this parameter is whether the BME680 immediately starts a measurement cycle or whether it stays in sleep mode after its initialization.

If we would have the API as suggested with
```
int bme680_read(bme680_t *dev, struct bme680_field_data *data);
int bme680_start_measurement(bme680_t *dev);
int bme680_data_ready(bme680_t *dev);
```
the `power_mode` wouldn't be required.